### PR TITLE
Plans: Add Scan Product to the plans page

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request:
* Refactors the way that the plans page displays the plans page. 
* Adds a single array that is defined in PHP that then gets used to display the products on the plans page. 
* This hopeful make for a smaller JS bundle size. since the prodct definitions don't live in the JS. 
* It make it much easier to add new products down the line. By just adding to the array. 
* It also make it possible easier for different hosts/ partners to offer different products. The products array can be filterable. 

New Plans Page. 
After the change the pages look like this. 

When the site has no products:
Screen Shot 2020-04-28 at 1 05 32 PM

When the site has Jetpack Scan product and the Jetpack Premium. 
Screen Shot 2020-04-28 at 12 55 42 PM

The new pop up that was added in https://github.com/Automattic/jetpack/pull/15558 after the rebase.
Screen Shot 2020-04-28 at 12 52 08 PM
 
#### Testing instructions:
Visit the Plans page. 
- Does it work for you as before. 
- Does it look like before?
There should be no changes in that regard (expect for the new Scan Product)
- Do you see the new Scan Product. 

#### Proposed changelog entry for your changes:
* Adds the new Scan Product to the Plans page.

### CODE REVIEW

**Comment by rcanepa** [comment]
I don't feel comfortable enough to approve a PR like this yet but I tested it anyways. This is what I saw:

- I could upgrade my plan and buy products without problems.
- The UI looks fine on desktop and on mobile. I didn't see any change from the last version.
- From the code standpoint, it looks like a great improvement because the new approach is more generic and reusable, and we are shipping less code.

Awesome work @enejb !

**Comment by enejb** [comment]
@jeherve I don't think we will be ready to ship Jetpack Scan in 8.5 but the change to remove Jetpack Scan Product will just be a matter or removing it from the jetpack class file.

**Review by jeherve — CHANGES_REQUESTED** [review]
This seems to work well overall. I only have a few questions


*********

Is the current layout temporary?
image

I find it a bit odd to have a radio button in an area where I don't really have to choose between different solutions.

********

What would you think about adding an additional monthly/yearly picker at the bottom of the products? Right now it's kind of hard to switch when looking at Scan since you have to scroll to see one or the other:

image

File: class.jetpack.php
Comment (enejb) [reply class.jetpack.php:7329]: I don't think this list should come from the api. 
The main reason being is that the each Jetpack Should version should know what features (product) it should be selling. 

I know a lot of those features are moving to the cloud but there is still some product features code in there. 

So it make sense to me that the product should be made available in here.

**Comment by enejb** [comment]
> Is the current layout temporary?
I don't think so. The layout is consistent with the search one. I am also not a big fan of the radio button for the single feature. But that is the design that we have come up with :)

I don't think Scan should make it into the next version of Jetpack (8.5) but removing it will be as just removing it from the array. 

> What would you think about adding an additional monthly/yearly picker at the bottom of the products?

I think that could easily be done. But lets do it in a different PR if we get approval from design :) 

cc: @keoshi, @eeeeevon13  :)

File: class.jetpack.php
Comment (jeherve) [reply class.jetpack.php:7329]: > I don't think this list should come from the api.

I moved this discussion here so others can chime in:

p1HpG7-8ZR-p2

**Comment by enejb** [comment]
I talked to @jeherve and there is no design review needed since the design is staying the same. 
There is no hold on this since the PR is already pretty big and is holding us up from testing Jetpack Scan.

**Review by rcanepa — DISMISSED** [review]
> Visit the Plans page.
> 
> - Does it work for you as before.
> - Does it look like before?
> - There should be no changes in that regard (expect for the new Scan Product)
> - Do you see the new Scan Product.

I reviewed this PR again and these things work as expected.

**Review by jeherve — DISMISSED** [review]

**Comment by jeherve** [comment]
This should be good to merge after a rebase!

**Review by keoshi — DISMISSED** [review]
This looks good to me!

The only thing I'd note is that we're repeating the promo element inside the cards, making the entire layout more confusing than it should be since we can probably move that common element outside. That can surely be handled in a different PR, but the visuals on this one might help with this decision. cc @folletto @robertbpugh

**Comment by robertbpugh** [comment]
I agree it is getting busy with all the promos, so we may want to explore a different design for that. But we can't move it outside, as Search has no discount, the Backup discount is still 70%, and Scan is 30% (the price is $7 / $70 as noted here under the pricing table: https://jetpackp2.wordpress.com/2020/04/02/jetpack-scan-pricing/ )

**Review by jeherve — APPROVED** [review]

**Pull request merged**
